tests/nested/manual/fde-on-classic/mk-image.sh

```diff
@@ -0,0 +1,351 @@
+        if [ "${IN_SNAP_F[$snap_n]}" != "" ]; then
+            IS_UNASSERTED[$i]=true
+            SNAP_F[$i]=${IN_SNAP_F[$snap_n]}
+            SNAP_P[$i]=${IN_SNAP_P[$snap_n]}
+        else
+            IS_UNASSERTED[$i]=false
+            SNAP_F[$i]=$(find "$CACHE" -maxdepth 1 -name "${SNAP_NAME[$i]}_*.snap" -printf "%f\n")
+            SNAP_P[$i]="$CACHE/${SNAP_F[$i]}"
+        fi
+    done
+
+    # Copy base filesystem
+    sudo tar -C "$DESTDIR" -xf "$CACHE"/ubuntu-base-22.04-base-amd64.tar.gz
+
+    # Create basic devices to be able to install packages
+    [ -e "$DESTDIR"/dev/null ] || sudo mknod -m 666 "$DESTDIR"/dev/null c 1 3
+    [ -e "$DESTDIR"/dev/zero ] || sudo mknod -m 666 "$DESTDIR"/dev/zero c 1 5
+    [ -e "$DESTDIR"/dev/random ] || sudo mknod -m 666 "$DESTDIR"/dev/random c 1 8
+    [ -e "$DESTDIR"/dev/urandom ] || sudo mknod -m 666 "$DESTDIR"/dev/urandom c 1 9
+    # ensure resolving works inside the chroot
+    echo "nameserver 8.8.8.8" | sudo tee -a "$DESTDIR"/etc/resolv.conf
+    # install additional packages
+    sudo chroot "$DESTDIR" /usr/bin/sh -c "DEBIAN_FRONTEND=noninteractive apt update"
+    local pkgs="snapd ssh openssh-server sudo iproute2 iputils-ping isc-dhcp-client netplan.io vim-tiny kmod cloud-init"
+    sudo chroot "$DESTDIR" /usr/bin/sh -c \
+         "DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends -y $pkgs"
+    # netplan config
+    cat > "$CACHE"/00-ethernet.yaml <<'EOF'
+network:
+  ethernets:
+    any:
+      match:
+        name: e*
+      dhcp4: true
+  version: 2
+EOF
+    sudo cp "$CACHE"/00-ethernet.yaml "$DESTDIR"/etc/netplan
+
+    # mount bits needed to be able to update boot assets
+    sudo mkdir -p "$DESTDIR"/boot/grub "$DESTDIR"/boot/efi
+    sudo tee "$DESTDIR"/etc/fstab <<'EOF'
+/run/mnt/ubuntu-boot/EFI/ubuntu /boot/grub none bind 0 0
+EOF
+
+    # ensure we can login
+    sudo chroot "$DESTDIR" /usr/sbin/adduser --disabled-password --gecos "" user1
+    printf "ubuntu\nubuntu\n" | sudo chroot "$DESTDIR" /usr/bin/passwd user1
+    echo "user1 ALL=(ALL) NOPASSWD:ALL" | sudo tee -a "$DESTDIR"/etc/sudoers
+
+    # set password for root user
+    sudo chroot "$DESTDIR" /usr/bin/sh -c 'echo root:root | chpasswd'
+    sudo tee -a "$DESTDIR/etc/ssh/sshd_config" <<'EOF'
+PermitRootLogin yes
+PasswordAuthentication yes
+EOF
+
+    # Populate snapd data
+    cat > modeenv <<EOF
+mode=run
+recovery_system=$SEED_LABEL
+current_recovery_systems=$SEED_LABEL
+good_recovery_systems=$SEED_LABEL
+base=${SNAP_F[base]}
+gadget=${SNAP_F[gadget]}
+current_kernels=${SNAP_F[kernel]}
+model=canonical/ubuntu-core-22-pc-amd64
+grade=dangerous
+model_sign_key_id=9tydnLa6MTJ-jaQTFUXEwHl1yRx7ZS4K5cyFDhYDcPzhS7uyEkDxdUjg9g08BtNn
+current_kernel_command_lines=["snapd_recovery_mode=run console=ttyS0 console=tty1 panic=-1 quiet splash"]
+EOF
+    sudo cp modeenv "$DESTDIR"/var/lib/snapd/
+    # needed from the beginning in ubuntu-data as these are mounted by snap-bootstrap
```

Comment:
(nitpick) we have `snap prepare-image --classic` for classic+modes now so this can probably be simplified but that should be a followup.

Reply:
I can leave a TODO I suppose